faults/common.go

```diff
@@ -0,0 +1,124 @@
+// Package faults provides common headers and client-side fault injection functionality.
+package faults
+
+import (
+	"fmt"
+	"log/slog"
+	"math/rand/v2"
+	"strconv"
+	"strings"
+	"time"
+)
+
+type GetHeaderFn func(key string) string
+type ResumeFn func() (interface{}, error)
+type ResponseFn func(code int, message string) (interface{}, error)
+type SleepFn func(d time.Duration)
+
+// Object to ensure a random number is only generated at most 1 time.
+type randSingleton struct {
+	randInt *int
+}
+
+func (r randSingleton) getRandInt() int {
+	if r.randInt == nil {
+		*r.randInt = rand.IntN(100)
+	}
+	return *r.randInt
+}
```

Comment:
use [`sync.OnceValue`](https://pkg.go.dev/sync#OnceValue) instead?

Reply:
the current implementation is definitely not thread-safe.

Reply:
Updated.